addons/pos_pine_labs/static/src/app/utils/payment/payment_pine_labs.js

```diff
@@ -0,0 +1,221 @@
+import { _t } from "@web/core/l10n/translation";
+import { PaymentInterface } from "@point_of_sale/app/payment/payment_interface";
+import { AlertDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { serializeDateTime } from "@web/core/l10n/dates";
+import { offlineErrorHandler } from "@point_of_sale/app/errors/error_handlers";
+import { register_payment_method } from "@point_of_sale/app/store/pos_store";
+
+const REQUEST_TIMEOUT_MS = 5000;
+const CANCEL_REQUEST_TIME_LIMIT_MS = 125000;
+const { DateTime } = luxon;
+
+export class PaymentPineLabs extends PaymentInterface {
+    setup() {
+        super.setup(...arguments);
+        this.pollingTimeout = null;
+        this.inactivityTimeout = null;
+        this.payment_stopped = false;
+    }
+
+    send_payment_request(cid) {
+        super.send_payment_request(cid);
+        return this._processPineLabs();
+    }
+
+    pendingPineLabsPaymentLine() {
+        return this.pos.getPendingPaymentLine("pine_labs");
+    }
+
+    send_payment_cancel(order, cid) {
+        super.send_payment_cancel(order, cid);
+        return this._pineLabsCancel();
+    }
+
+    _callPineLabs(data, action) {
+        return this.pos.data
+            .call("pos.payment.method", action, [[this.payment_method_id.id], data])
+            .catch((error) => {
+                const line = this.pendingPineLabsPaymentLine();
+                if (line) {
+                    line.set_payment_status("force_done");
+                }
+                offlineErrorHandler(this.env, error, error);
+            });
+    }
+
+    /**
+     * This method processes the response from Pine Labs
+     * when we request payment or cancellation.
+     */
+    async _pineLabsHandleResponse(response) {
```

Comment:
```suggestion
    async _pineLabsHandleResponse(response) {
```

don't overload this method. It would better if you have handlers for the 3 cases: making request, cancelling and fetching status.